react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,31 @@
+
+import React from 'react';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { count: 0 };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({ count: this.state.count + 1 });
+  }
+
+  render() {
+    if (this.state.count <= 3) {
+      return <button onClick={this.handleClick} className="hot-button btn3">Hot Button!</button>;
+    } else if (this.state.count <= 6) {
+      return <button onClick={this.handleClick} className="hot-button btn6">Hot-button!</button>;
+    } else if (this.state.count <= 9) {
+      return <button onClick={this.handleClick} className="hot-button btn9">Hot-button!</button>;
+    } else if (this.state.count <= 12) {
+      return <button onClick={this.handleClick} className="hot-button btn12">Hot-button!</button>;
+    } else if (this.state.count <= 15) {
+      return <button onClick={this.handleClick} className="hot-button btn15">Hot-button!</button>;
+    } else if (this.state.count >= 16) {
+      return <button onClick={this.handleClick} className="hot-button btn18">Hot-button!</button>;
+    }
```

Comment:
This is a lot of repetition and would be hard to maintain. A better approach would be to use the if statements to determine the correct class to use, then return a single button at the end.
```JSX
let btnClass = '';

if (this.state.count <= 3) {
  btnClass = "btn3";
} else if (this.state.count <= 6) {
  btnClass = "btn6";
} else if (this.state.count <= 9) {
  btnClass = "btn9";
} else if (this.state.count <= 12) {
  btnClass = "btn12";
} else if (this.state.count <= 15) {
  btnClass = "btn15";
} else if (this.state.count >= 16) {
  btnClass = "btn18";
}

return <button onClick={this.handleClick} className={`hot-button ${btnClass}`}>Hot-button!</button>;
```